[Revamp Landing Screen] Jetpack Landing Screen design tweaks

ovitrif opened the pull request:
Fixes #17214

Implements the latest design updates to the Jetpack Landing Screen. These are only UI tweaks to different elements, with no updates to the animation.

There's also a few code cleanup changes done to tidy up the Compose code. One particular change is that private composable methods lose their `modifier` parameter since these aren't to be called outside their single-use parent composable.

I've also changed the way the large texts are getting built and colored, making use of Compose's `AnnotatedString`. This was done to have a better control over the line height, since in the latest design update it was shrunk considerably, and by using a `Text` composable for each line of text it wasn't easy and clean to achieve the desired result.

To test:

<details>
<summary><b>Prerequisite</b> - Enable the <code>LandingScreenRevamp</code> feature flag</summary>

- Either `login` → `Me` → `App Settings` → `Debug Settings` → enable `LandingScreenRevamp` → `RESTART THE APP` → `logout` or
- Apply this hard-coded solution:

<details>
<summary>Patch to enable the <code>LANDING_SCREEN_REVAMP</code> build flag</summary>

```diff
diff --git a/WordPress/build.gradle b/WordPress/build.gradle
index 63260a4ca8..fe999fa92f 100644
--- a/WordPress/build.gradle
+++ b/WordPress/build.gradle
@@ -112,7 +112,7 @@ android {
         buildConfigField "boolean", "READER_COMMENTS_MODERATION", "false"
         buildConfigField "boolean", "SITE_INTENT_QUESTION", "true"
         buildConfigField "boolean", "SITE_NAME", "false"
-        buildConfigField "boolean", "LANDING_SCREEN_REVAMP", "false"
+        buildConfigField "boolean", "LANDING_SCREEN_REVAMP", "true"
         buildConfigField "boolean", "LAND_ON_THE_EDITOR", "false"
         buildConfigField "boolean", "BLOGGING_PROMPTS", "false"
         buildConfigField "boolean", "QUICK_START_EXISTING_USERS_V2", "false"
```  
</details>

---

</details>

##### 1️⃣ Test Case 1: Happy Flow

0. Open the JP app and make sure you're logged out
1. **Verify** That the _UI_ looks similar to the design specs:

<img width="679" alt="Screenshot 2022-09-29 at 13 50 10" src="https://user-images.githubusercontent.com/4588074/193023847-73c11189-545d-4a19-b7bb-406f7770f2e3.png">

2. Toggle dark mode on the device
3. **Verify** That the _UI on dark mode_ looks similar to the dark mode design specs.

## Regression Notes
1. Potential unintended areas of impact
   None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
   Manual testing, though no regression expected - the PR is adjusting the UI of a recently added screen that's behind a feature flag.

4. What automated tests I added (or what prevented me from doing so)
   N/a

## Previews

| Dark Mode | Light Mode |
| - | - |
| <img width="314" src="https://user-images.githubusercontent.com/4588074/193025210-beb0e66b-8e1e-4742-b7ba-2c7c03033a11.png"> | <img width="314" src="https://user-images.githubusercontent.com/4588074/193025187-b7422acf-7c04-4d3d-bd98-c5cf8e9f676f.png"> |

### 🎬 Video Preview
https://user-images.githubusercontent.com/4588074/193025599-2c254878-2ff2-452f-9a8a-adc4acf962d6.mp4

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ovitrif committed:
Tweak color of large texts

ovitrif committed:
Tweak spacing and size of large texts

ovitrif committed:
Use annotated string to color large texts

ovitrif committed:
Add preview for LoopingTextWithBackground

ovitrif committed:
Tweak frosted glass effect

ovitrif committed:
Tweak top linear gradient

ovitrif committed:
Add preview for top linear gradient

ovitrif committed:
Update login screen bg color

ovitrif committed:
Add dark preview for looping text with background

ovitrif committed:
Apply alpha to blob bg image and looping text

ovitrif committed:
Update login screen buttons style

ovitrif committed:
Remove duplicate alpha of frosted glass column bg

ovitrif committed:
Remove unused Float.isNegative

ovitrif committed:
Fix alpha and padding on looping text

antonis reviewed (approved):
Great work @ovitrif 👍 
the code changes look consistent to me and the design matches the desired 🎉 

Edit: Tested both on Pixel 5 (Android 13) and Huawei P9 (Android 7)

antonis commented:
Btw the changes on this PR seem to fix [top/bottom bar issue reported before](https://github.com/wordpress-mobile/WordPress-Android/pull/17117#issuecomment-1253789439). I'm still able to reproduce [it on the parent PR](https://github.com/wordpress-mobile/WordPress-Android/pull/17207).

ovitrif commented:
@shaunandrews I added you as reviewer for the design tweaks, you can download the link to the Jetpack App from here:

https://github.com/wordpress-mobile/WordPress-Android/pull/17231#issuecomment-1262193829

Note that since we have to be logged out to see the screen, but logged in to toggle the feature flag, the flow to toggle the feature flag that enables this screen is a bit counter-intuitive but doable. It's explained in the **Prerequisites** collapsible panel of the description, or for TL;DR; here:

> `login` → `Me` → `App Settings` → `Debug Settings` → enable `LandingScreenRevamp` → RESTART THE APP → logout.

ovitrif committed:
Fix off-by-one error in large text coloring

ovitrif commented:
Just figured out I've done the infamous [off-by-one](https://user-images.githubusercontent.com/4588074/193038544-4fb67f26-1843-4b0e-a111-0ac366799f90.png) programming error when coloring the large texts, a fix was pushed in https://github.com/wordpress-mobile/WordPress-Android/pull/17231/commits/fc840a6b3302d7e074cbc441083cde34e7c1aaf2:

| Before | After |
| - | - |
| <img width="314" src="https://user-images.githubusercontent.com/4588074/193038544-4fb67f26-1843-4b0e-a111-0ac366799f90.png"> | <img width="314" src="https://user-images.githubusercontent.com/4588074/193038172-db4306e8-9404-48c6-901c-4442a329fcb0.png"> |

**Notice** the different color for the "Add an author" text, and inverted even / odd colors that are now matching the design.

ovitrif commented:
Moved this PR to the next milestone: `20.9`, since getting feedback on the design implementation is quite important on this one.

shaunandrews commented:
Really great to see this come to life on my phone. Nice work.

I do have a few things I think need to be address:

- Holding my phone in a "normal" position (Say, between 45° and 60°) the speed of the scrolling feels a little too fast. Can we slow it down a bit so it's easier to read.
- Some of the longer lines, like "Fix a security issue," wrap to multiple lines as a **paragraph**. The line-height of this **paragraph** needs to be smaller, somewhere around 105%. This is different from the space between each **paragraph**, which in the Figma is listed as 5px. I know these values might not map to the code, and that there are some limitations here. At the very least, we need to adjust the line-height of the **paragraphs** to be smaller than the space between the **paragraphs** — otherwise it becomes difficult to discern related lines of text.
- The physics need to be more responsive. As I tilt my device around, I expect the scrolling velocity (speed+direction) to react quicker than it currently does. Right now it feels like tepid honey, but it should feel more like warm cream.

--

These last two items are more wishful thinking, than suggestions for this specific PR:

- I really, really want to scroll the text manually with my finger. I know I said we didn't need it in the past, and we still don't — but I still find myself trying to scroll with my finger and being disappointed when it doesn't do anything. It could be fun to have a rubberband-like affect, where after manually scrolling the list bounces back to its normal behavior.
- It could be pretty cool to add some haptic feedback to the scrolling. I can imagine a soft tap for each line that scrolls by; It would be fun to feel the tapping speed up and down as I move my device around.

iamthomasbishop commented:
@ovitrif I wanted to share some feedback that I had while chatting with @shaunandrews and Paul F about this screen. I thought there could be a stronger color-contrast between primary/secondary labels so I tried a couple of things. My initial suggestion was to use the same color for both labels (base of `JP Green 50` in light, `JP Green 20` in dark), then use opacity to de-emphasize the secondary label a bit ([example](https://user-images.githubusercontent.com/1200432/193340784-e8cbdbb1-82de-4989-81b2-7b81018c0c67.png)). However, after discussing with the others, we agreed that the current dark theme implementation (`JP Green 50`, `JP Green 20`) is probably okay as-is, but something was off with the light theme. For that, we could try `JP Green 50` for both labels, and apply ~50% opacity to the secondary label. Here's what that looks like side-by-side:

![image](https://user-images.githubusercontent.com/1200432/193341186-ddd562d6-a11c-4a4c-9399-d5b66167e001.png)

Does that sound good? Or should we apply the same approach (uniform label color, varying opacity) to both themes?

AliSoftware commented:
Moving this PR to milestone 21.0 as I'm freezing the code for 20.9.

@ovitrif Note: I see that this PR doesn't seem to have a CHANGELOG entry. Not sure if it's intentional (i.e. you don't think those changes are worth mentioning to end users) or if you just forgot, so, in doubt, I preferred to ask 🙂

ovitrif commented:
Many thanks Shaun & Thomas for the taking time to provide your feedback on this PR ❤️ !

> - Holding my phone in a "normal" position (Say, between 45° and 60°) the speed of the scrolling feels a little too fast. Can we slow it down a bit so it's easier to read.
> - Some of the longer lines, like "Fix a security issue," wrap to multiple lines as a **paragraph**. The line-height of this **paragraph** needs to be smaller, somewhere around 105%. This is different from the space between each **paragraph**, which in the Figma is listed as 5px. I know these values might not map to the code, and that there are some limitations here. At the very least, we need to adjust the line-height of the **paragraphs** to be smaller than the space between the **paragraphs** — otherwise it becomes difficult to discern related lines of text.
> - The physics need to be more responsive. As I tilt my device around, I expect the scrolling velocity (speed+direction) to react quicker than it currently does. Right now it feels like tepid honey, but it should feel more like warm cream.

@shaunandrews I agree with each point and I'll look into the adjusting the implementation to consider each of these 👍 

As for the last two points (that are not quoted) most probably we won't include these in the current iteration, but we'll add them as separate tasks that we can look into or further updates!

> Does that sound good? Or should we apply the same approach (uniform label color, varying opacity) to both themes?

Sounds good to me @iamthomasbishop 🙇 , thank you for these remarks. I feel it's doable with applying the different color & varying opacity as described, no specific need to go with the same approach for both themes. I'll make an update and post previews here so that we can get a clear idea of what's looking best 👍 .

> Moving this PR to milestone 21.0 as I'm freezing the code for 20.9.
> 
> @ovitrif Note: I see that this PR doesn't seem to have a CHANGELOG entry. Not sure if it's intentional (i.e. you don't think those changes are worth mentioning to end users) or if you just forgot, so, in doubt, I preferred to ask 🙂

Thank you @AliSoftware for updating the milestone, it makes sense 👍 .

As for adding a changelog entry, these changes are just small UI tweaks which most probably won't be reflected on a changelog 👍

ovitrif committed:
Lower max scroll speed

ovitrif committed:
Increase scroll sensor sensitivity

ovitrif committed:
Increase scroll responsiveness to sensor values

ovitrif committed:
Make sure drift doesn't change reversing of scroll direction

ovitrif commented:
@shaunandrews I've adjusted the scroll responsiveness and speed, targeting these 2 points:

> - Holding my phone in a "normal" position (Say, between 45° and 60°) the speed of the scrolling feels a little too fast. Can we slow it down a bit so it's easier to read.
> - The physics need to be more responsive. As I tilt my device around, I expect the scrolling velocity (speed+direction) to react quicker than it currently does. Right now it feels like tepid honey, but it should feel more like warm cream.

Can you please give it another check and share your feedback?
This comment is updated with a link to an app that contains the latest changes: https://github.com/wordpress-mobile/WordPress-Android/pull/17231#issuecomment-1262193829

It would be great to also get feedback from @mkevins on how the code changes feel as a solution for the 2 requested points quoted above, as my understanding for the physics model is probably not complete 🙇

mkevins commented on WordPress/src/jetpack/java/org/wordpress/android/ui/accounts/login/LoginPrologueRevampedViewModel.kt:
```diff
@@ -73,7 +73,9 @@ class LoginPrologueRevampedViewModel @Inject constructor(
         override fun onSensorChanged(event: SensorEvent?) {
             event?.values?.let { (_, yAxisAcceleration, _) ->
-                acceleration = yAxisAcceleration * ACCELERATION_FACTOR + DRIFT
+                acceleration = yAxisAcceleration * ACCELERATION_FACTOR +
+                        (if (yAxisAcceleration >= -0.2f) DRIFT else -DRIFT) // drift shouldn't change scroll direction.
+
```
Can you elaborate on the goal of this change?

The original purpose I had in mind for the `DRIFT` constant was to provide a slight "force" (i.e. biasing the acceleration) so that when the y component of the acceleration vector was zero (i.e. device flat on a table), the text would still scroll.

Is the intention of this change to remove the stability region when the acceleration exactly "cancels" the drift (e.g. when the device is in a "declined" position, tilted slightly backward)? Afaict, this change introduces a discontinuity in the output for a continuous input.

From earlier discussions, I'm thinking that this wasn't the most appropriate place to put this `DRIFT`, and that rather than baking it into the `onSensorChanged` function, it would be better to be moved to the `updateForFrame` function, which is more concerned with updating the physics model. Wdyt?

ovitrif replied:
> Afaict, this change introduces a discontinuity in the output for a continuous input.

Good point! this makes it far from ideal but with the latest adjustments to `ACCELERATION_FACTOR` & `VELOCITY_DECAY` it seemed like a needed thing.

The picture below illustrates the issue it's fixing:

<img width="813" alt="pose_motion_dir" src="https://user-images.githubusercontent.com/4588074/193809700-721871b4-811d-41a5-8f9d-26388bdc6d7e.png">

At the device pose from the image we'd expect the scroll direction to be reversed, this was not the case since the `DRIFT` is a constant value added on top of the sensor data.

To explain a bit more why I adjusted the acceleration factor and velocity decay we have to take into consideration the `MAXIMUM_VELOCITY`.
This value is constraining the "responsiveness" of the physics, since any device rotation that outputs a value outside of this **max limit**, either positive or negative, will not be reflected on the scrolling speed.

before starting to make adjustments, I've added logging for the `velocity` and played with the `Virtual sensors` → `Device Pose` on the emulator.
This way I noticed with numbers what felt to be the visible with the eye: the device rotation is only affecting the speed of the animation when we're almost near the "flat on the table" device pose. Anything above this has no effect on the speed.

<img width="813" alt="Screenshot 2022-10-04 at 13 49 21" src="https://user-images.githubusercontent.com/4588074/193812156-551d47a5-40c1-4531-b577-844da3429514.png">

As seen above, any vertical rotation above that pose has no effect on the animation speed.

While playing with the values I figured out lowering the `ACCELERATION_FACTOR` results in more responsiveness, I.e. The rotation threshold needed to hit the `MAXIMUM_VELOCITY` can be adjusted so that we have different `velocity` over the entire rotation range from full vertical to full horizontal:

<img width="1155" alt="Screenshot 2022-10-04 at 13 55 22" src="https://user-images.githubusercontent.com/4588074/193813316-f63976d3-a30b-46d4-9516-a905e4f187c2.png">

In the image above we have the current values `ACCELERATION_FACTOR = -0.015f` instead of `-0.1f`.
Because of this, the velocity starts to change from a rotation of `-18` compared to `-83`. This translates into getting the feeling that the device rotation has more effect on the animation speed, thus imho making it feel more "responsive".

mkevins replied:
> At the device pose from the image we'd expect the scroll direction to be reversed, this was not the case since the `DRIFT` is a constant value added on top of the sensor data.

Ah, this makes sense in light of the changes to acceleration that you described. In that case, `DRIFT` became proportionally larger as the acceleration factor was reduced. Honestly, the `DRIFT` is probably the last thing I'd add, as it isn't too important.. maybe it's simpler to just delete it and add it back at the end.

> To explain a bit more why I adjusted the acceleration factor and velocity decay we have to take into consideration the MAXIMUM_VELOCITY.
This value is constraining the "responsiveness" of the physics, since any device rotation that outputs a value outside of this max limit, either positive or negative, will not be reflected on the scrolling speed.

Right, this makes sense, and why I suggested removing the maximum velocity while playing with the balance of acceleration and friction. But, on further thought, if the maximum velocity will be eventually constrained to such a degree, I think the general issue is that we have very little to work with in terms of expressiveness, unfortunately. The reason for this is that the relative changes in velocity will only ever be _very small_: from barely moving downward to barely moving upward, to barely moving in between :sweat_smile: . And, that would be from tilting the device in extreme positions to achieve very little movement.

> Good point! this makes it far from ideal but with the latest adjustments to ACCELERATION_FACTOR & VELOCITY_DECAY it seemed like a needed thing.

To clarify, when I mentioned the discontinuity, I didn't mean to raise that as an issue / problem, but instead to describe the change in mathematical relationship resulting from those changes. For example, it may very well be _desirable_ to have this discontinuity - I guess it depends on the goal we are trying to achieve. :smile: 

Since this model originally assumed a somewhat larger maximum velocity, and since a degree of expressiveness was possible within that range, which is no longer possible with a lower velocity, it may be worth considering a model without any inertia at all.

I've implemented a different approach on this branch: https://github.com/wordpress-mobile/WordPress-Android/compare/issue/17214-final-design-touces...issue/17214-final-design-touces--orientation-angle-approach, which utilizes the orientation calculations. In this model, there is no acceleration, and instead, the velocity is directly mapped to the pitch angle of the device. Removing the inertia from the model may help to make the changes to the animation feel more "instantaneous" in response to device movements. Wdyt?

Also, in that branch, I removed the `DRIFT` value, since it was never really part of the original designs, so it seems unnecessary, but this can be also reverted / tweaked, depending on what we want the behavior to be. Generally, I aimed in this branch to reduce the number of parameters so that the fine-tuning process could be faster (a smaller "search space" for the ideal values :smiley: ).

ovitrif replied:
Thanks a lot for exploring the different approach for sensors motion @mkevins 🙇 

I opened a PR targeting this branch from your work 🎉  : https://github.com/wordpress-mobile/WordPress-Android/pull/17265

ovitrif replied:
The PR https://github.com/wordpress-mobile/WordPress-Android/pull/17265 was merged, improving the responsiveness of the animation to sensors.

mkevins commented:
Hi @ovitrif :wave: :smiley: 

Thank you for your updates!

> It would be great to also get feedback from @mkevins on how the code changes feel as a solution for the 2 requested points quoted above, as my understanding for the physics model is probably not complete

Generally speaking, I can recommend that we can make the physics more "responsive" by _increasing_ the magnitude acceleration factor (but keeping it negative, since the screen offset coordinates are expressed as an inverted y-axis). I would start with balancing the larger acceleration (which effectively makes the "mass" of the moving view "lighter" and more quick to respond to orientation changes), and the velocity decay (which acts sort of like a force of friction). I would probably try to play with these with a large maximum velocity, and then decrease the maximum to fine tune the velocity in the most extreme positions (i.e. completely vertical).

#### Some issues I've encountered

At some point, after the last few changes, some issues seem to have been introduced, though I haven't tracked down which commits are responsible:

* The frosted glass seems very opaque (it almost seems like a plain white background now)
* The words that do appear behind the frosted glass seem to have a different padding then the unblurred words
* The performance seems to be suffering a bit (I'm not exactly sure why, though I have some theory)
  * On my Pixel 3a (physical device) the animation seems to be a bit "jumpy", like the frames cannot keep up

Regarding the performance, I did not observe any performance issues when testing out the branch from this PR: https://github.com/wordpress-mobile/WordPress-Android/pull/17207, so I did a `git bisect` between the `HEAD` here and there and found that the commit that seems to introduce the performance issues is this one: https://github.com/wordpress-mobile/WordPress-Android/pull/17231/commits/5eefdd61d1495a014fa1eefdf3b6970edbf0a754. I'm not quite sure why this change is so costly from a performance perspective, but it may have to do with the way the `alpha` modifiers are implemented in the `graphicsLayer`:

> Draw content with modified alpha that may be less than 1.
Usage of this API renders this composable into a separate graphics layer. Note when an alpha less than 1.0f is provided, contents are implicitly clipped to their bounds. This is because an intermediate compositing layer is created to render contents into first before being drawn into the destination with the desired alpha. This layer is sized to the bounds of the composable this modifier is configured on, and contents outside of these bounds are omitted.

Perhaps somehow this additional rendering is enough to make the performance suffer :man_shrugging: . I tried reverting those changes with the following patch:

<details>
<summary>Patch to revert alpha modifiers</summary>

```diff
diff --git a/WordPress/src/jetpack/java/org/wordpress/android/ui/accounts/login/components/LoopingTextWithBackground.kt b/WordPress/src/jetpack/java/org/wordpress/android/ui/accounts/login/components/LoopingTextWithBackground.kt
index e5b60c0e7e..04afd9093a 100644
--- a/WordPress/src/jetpack/java/org/wordpress/android/ui/accounts/login/components/LoopingTextWithBackground.kt
+++ b/WordPress/src/jetpack/java/org/wordpress/android/ui/accounts/login/components/LoopingTextWithBackground.kt
@@ -8,7 +8,6 @@ import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.foundation.layout.padding
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.draw.alpha
 import androidx.compose.ui.layout.ContentScale
 import androidx.compose.ui.res.colorResource
 import androidx.compose.ui.res.painterResource
@@ -30,12 +29,10 @@ fun LoopingTextWithBackground(
                 contentDescription = stringResource(R.string.login_prologue_revamped_content_description_bg),
                 contentScale = ContentScale.FillBounds,
                 modifier = Modifier.matchParentSize(),
-                alpha = 0.8f
         )
         LoopingText(
                 modifier = Modifier
                         .fillMaxSize()
-                        .alpha(0.8f)
                         .padding(horizontal = 20.dp)
                         .then(textModifier)
         )
```
</details>

and the performance was smooth again. Is there another way to achieve what we're after with these changes without resorting to using alpha here? For example, I wonder whether the raster image itself could be adjusted instead :thinking:

ovitrif committed:
Fine-tune ACCELERATION_FACTOR to hit max value on vertical position

ovitrif commented:
@mkevins Thank you for your review.

Regarding the adjustments to the physics model, I've explained a bit the reasoning behind my changes in this comment: https://github.com/wordpress-mobile/WordPress-Android/pull/17231#discussion_r986779157

> * The frosted glass seems very opaque (it almost seems like a plain white background now)

This is probably the case because of the higher value for blur radius (30dp) and the removal of additional alpha (previously there was a mistake in the fact that we've had both alpha from the BG color of the panel behind the buttons and another alpha layer of `60%` via the modifier).

We could tweak the blur radius to be lower, I picked `30` because that's the value in Figma.

> * The words that do appear behind the frosted glass seem to have a different padding then the unblurred words

The blur radius might be the reason it appears to be so. If I remove it, the padding seems to be the same: 

<details>
<summary>Preview without blur</summary>

<img width="314" src="https://user-images.githubusercontent.com/4588074/193849631-9f8c559f-b46d-40b1-ab45-734485211b92.png">

</details>


> * The performance seems to be suffering a bit (I'm not exactly sure why, though I have some theory)
On my Pixel 3a (physical device) the animation seems to be a bit "jumpy", like the frames cannot keep up
> [...]
> Is there another way to achieve what we're after with these changes without resorting to using alpha here? For example, I wonder whether the raster image itself could be adjusted instead 🤔

I think we can achieve the same result by applying alpha differently. that's indeed on the image level. There's also alpha applied to the large text, I'll adjust this by applying it on the colors of the text instead.

In any case if the performance still suffers I'd say we can exclude these changes; they were just last-minute fine-tuning adjustments that I've noticed while inspecting the elements in Figma and comparing with the values in the implementation. I don't think there'll be that much difference to notice without the alphas, while performance feels more important here imho.

ovitrif committed:
Bake alpha in blob bg image

ovitrif committed:
Bake alpha in large text colors

ovitrif commented:
> I think we can achieve the same result by applying alpha differently. that's indeed on the image level. There's also alpha applied to the large text, I'll adjust this by applying it on the colors of the text instead.

@mkevins Did this in 2 commits:
- **BG Image:** https://github.com/wordpress-mobile/WordPress-Android/pull/17231/commits/53be00a5c302ad1293ec9150190c8d9a9e4bdaf5
- **Text color:** https://github.com/wordpress-mobile/WordPress-Android/pull/17231/commits/674a9825fd1ae140bb79af1134625707db8d51c1

ovitrif committed:
Adapt large text colors to different alpha per label

ovitrif committed:
Adapt large text colors to different alpha per label

ovitrif commented:
> we agreed that the current dark theme implementation (`JP Green 50`, `JP Green 20`) is probably okay as-is, but something was off with the light theme. For that, we could try `JP Green 50` for both labels, and apply ~50% opacity to the secondary label. Here's what that looks like side-by-side

@iamthomasbishop I applied these specs in commit https://github.com/wordpress-mobile/WordPress-Android/pull/17231/commits/c3453d510009e68923c95bda74c81f7959e841b7.

This is how it looks:

| Light | Dark |
| - | - |
| `JP Green 50 @ 50% alpha`, `JP Green 50` | `JP Green 50`, `JP Green 20` |
| <img width="314" src="https://user-images.githubusercontent.com/4588074/193870305-19fd60a0-7b2f-4d44-a5c8-fd571c1b0d76.png"> | <img width="314" src="https://user-images.githubusercontent.com/4588074/193870326-c6d78fc8-c177-4f86-b3f2-948f584d571a.png"> |

Let me know if I understood this correctly, and wdyt of how it looks 🙇 .

P.S. Notice that initially there was an 80% alpha applied to the entire text layer (influencing both primary and secondary colors). That is gone now because of the performance issues with applying an overall alpha (see https://github.com/wordpress-mobile/WordPress-Android/pull/17231#issuecomment-1267122908).

ovitrif commented:
> Some of the longer lines, like "Fix a security issue," wrap to multiple lines as a paragraph. The line-height of this paragraph needs to be smaller, somewhere around 105%. This is different from the space between each paragraph, which in the Figma is listed as 5px. I know these values might not map to the code, and that there are some limitations here. At the very least, we need to adjust the line-height of the paragraphs to be smaller than the space between the paragraphs — otherwise it becomes difficult to discern related lines of text.

@shaunandrews I've applied a different line-height to the wrapping text (commit https://github.com/wordpress-mobile/WordPress-Android/pull/17231/commits/e752f1c60544c9e1f82eac7458d78dfd66020be5) that is calculated to be 105% of the font size.

Here's the difference:

| Before | After |
| - | - |
| <img width="314" src="https://user-images.githubusercontent.com/4588074/193872976-3d66eee5-7e0f-43f6-b951-277fdae13c07.png"> | <img width="314" src="https://user-images.githubusercontent.com/4588074/193872953-d5ffac38-f366-40e7-add0-c9e3e25fc813.png"> |

Check the "Respond to Comments" wrapped text ;)

⚠️ **Notice** There may be a larger base line-height (that is the line height between the texts of different colors). That's not straightforward to adjust and we pretty much have to go with what we're getting since this is already the minimum value we can set to avoid other artifacts (like different space when the texts start repeating).

OTOH there's now a different, smaller line height for the wrapped texts.

ovitrif committed:
Merge remote-tracking branch 'origin/issue/17214-final-design-touces' into issue/17214-final-design-touces

# Conflicts:
#	WordPress/src/jetpack/res/values-night/colors.xml

ovitrif committed:
Use different line-height for wrapped texts

ovitrif commented:
This is ready for another pass 👍 , please check comments with mentions for the addressed issues.

iamthomasbishop commented:
> I applied these specs in commit c3453d5. This is how it looks:

Colors look great, thanks @ovitrif!

mkevins committed:
Adapt the landing screen animation to use orientation angle

This implements a different approach: updating the velocity value in the
animation directly from the pitch angle of the device. This removes all
inertia from the physics model, which may help to make the interaction
more responsive, given the narrow range of appropriate velocity values
we are constrained to.

mkevins committed:
Remove DRIFT parameter from animation model

Originally, the DRIFT parameter was used to provide movement for cases
where the device lacked sensor support. This was implemented
incorrectly, since it was only ever applied in the `onSensorChanged`
function, which never runs on such devices. It also had a side effect
which was not part of the original designs: it biased the interaction
such that it resulted in movement when the device was flat on a table.

Since this was not ever a requirement, it may be simpler to just remove
this parameter, since we can provide default behavior for devices
without sensor support instead via default sensor values (in this case,
a default value for the device pitch).

mkevins committed:
Make velocity variable local

shaunandrews commented:
Feeling great, especially with the physics changes in #17265

ovitrif committed:
Merge branch 'trunk' into issue/17214-final-design-touces

mkevins committed:
Update code comments to provide more details about the model

ovitrif committed:
Merge pull request #17265 from wordpress-mobile/issue/17214-final-design-touces--orientation-angle-approach

Revamp Jetpack Landing Screen - Use orientation angle approach for text scrolling speed

ovitrif commented:
I've merged the physics changes from https://github.com/wordpress-mobile/WordPress-Android/pull/17265 🎉 

@wordpress-mobile/manakin Please give this another check 🙇

ovitrif committed:
Merge remote-tracking branch 'origin/issue/17214-final-design-touces' into issue/17214-final-design-touces

ovitrif committed:
Merge branch 'trunk' into issue/17214-final-design-touces

ovitrif committed:
Fix bg color of WP landing screen on light theme

antonis reviewed (approved):
Great work @ovitrif 👍 
The app works great and the code looks consistent to me 🎉
